src/main/kotlin/org/rust/ide/inspections/lints/RsUnknownCrateTypesInspection.kt

```diff
@@ -0,0 +1,33 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.inspections.lints
+
+import com.intellij.psi.PsiElement
+import org.rust.ide.inspections.RsProblemsHolder
+import org.rust.lang.core.psi.RsLitExpr
+import org.rust.lang.core.psi.RsMetaItem
+import org.rust.lang.core.psi.RsVisitor
+import org.rust.lang.core.psi.ext.stringValue
+
+class RsUnknownCrateTypesInspection : RsLintInspection() {
+    override fun getLint(element: PsiElement): RsLint = RsLint.UnknownCrateTypes
+
+    override fun buildVisitor(holder: RsProblemsHolder, isOnTheFly: Boolean): RsVisitor =
+        object : RsVisitor() {
+            override fun visitLitExpr(o: RsLitExpr) {
+                val parent = o.parent as? RsMetaItem ?: return
```

Comment:
I think we should check only top level meta items not to produce false-positive errors in such cases
```rust
#[bar(crate_type = "bar")]
fn foo() {}
```
where `bar` is proc macro

Reply:
See `org.rust.lang.core.psi.ext.RsMetaItemKt#isRootMetaItem`, it should help here